### DISCUSSION TITLE
hasp/hasp01/command/p[1].b[6].txt causing 0x1a error.

### DISCUSSION THREAD

**Discussion by rebel7580** [HASwitchPlate/HASPone#91] [Q&A] [answered]
Just breadboarded a display hooked up to an NodeMCU for testing. Basics work.

espVersion:1.01
espCore:2_7_4
lcdVersion:3

All the documention says that to set text in a button, use something like the following (Page 1 selected):

  ```
topic: hasp/hasp01/command/p[1].b[6].txt   payload: Temp
```
I get error 0x1a when I send this.

```
[+26203.051s] MQTT IN: 'hasp/hasp01/command/p[1].b[6].txt' : 'Temp'
[+26203.061s] HMI OUT: 'p[1].b[6].txt=Temp'
[+26203.066s] HMI IN: [4 bytes]: 0x1a 0xff 0xff 0xff
[+26203.072s] HMI IN: [Invalid Variable name or attribute] 0x1a
[+26203.082s] MQTT OUT: 'hasp/hasp01/state/json' : '{"event":"nextion_return_data","return_code":"0x1a","return_code_description":"Invalid Variable name or attribute"}'
```

If I send the following instead:

```
  topic: hasp/hasp01/command   payload: b6.txt=Temp
```  
it works.

```
[+26220.535s] MQTT IN: 'hasp/hasp01/command' : 'b6.txt="Temp"'
[+26220.547s] HMI OUT: b6.txt="Temp"
```

Am I doing something wrong?

Thanks!

**Comment by aderusha** [comment, marked as answer]
String literals in Nextion need quotes around them.  `b6.txt="Temp"` will work,  but `b6.txt=Temp` will not.

**Reply by rebel7580** [reply]
That did it! 
Great Project!

Thanks.
